src/memory.rs

```diff
@@ -238,9 +238,9 @@ impl MemoryReference {
     /// info!("final position: {}", pos);
     /// creep.move_to(&pos);
     /// ```
-    pub fn get<T>(&self, key: &str) -> Result<Option<T>, ConversionError>
+    pub fn get<T, E>(&self, key: &str) -> Result<Option<T>, E>
```

Comment:
I think this could avoid the extra generic parameter if we just return `Result<Option<T>, <T as TryFrom<Value>>::Error>`. Since `Error` is an associated type, there's necessarily only one `Error` type per  `TryFrom<T>` implementation - so the `E` here would only ever have exactly one possible type.

Reply:
```suggestion
    pub fn get<T>(&self, key: &str) -> Result<Option<T>, <T as TryFrom<Value>>::Error>
```

I appreciate your work doing this! Only reason I suggest this is since I think it'd be a better solution. (but I really haven't been too involved in the project lately, thus I'm just commenting here rather than actually making the change)